src/particles/diagnostics/DiagnosticOutput.cpp

```diff
@@ -26,52 +27,66 @@ namespace impactx::diagnostics
-                amrex::ParticleReal const * const AMREX_RESTRICT part_pt = soa_real[RealSoA::pt].dataPtr();
+                        // preparing access to particle data: SoA of Reals
+                        auto &soa_real = pti.GetStructOfArrays().GetRealData();
+                        amrex::ParticleReal const *const AMREX_RESTRICT part_px = soa_real[RealSoA::ux].dataPtr();
+                        amrex::ParticleReal const *const AMREX_RESTRICT part_py = soa_real[RealSoA::uy].dataPtr();
+                        amrex::ParticleReal const *const AMREX_RESTRICT part_pt = soa_real[RealSoA::pt].dataPtr();
 
-                if (otype == OutputType::PrintParticles) {
-                    // print out particles (this hack works only on CPU and on GPUs with
-                    // unified memory access)
-                    for (int i=0; i < np; ++i) {
+                        if (otype == OutputType::PrintParticles) {
+                            // print out particles (this hack works only on CPU and on GPUs with
+                            // unified memory access)
+                            for (int i = 0; i < np; ++i) {
 
-                        // access AoS data such as positions and cpu/id
-                        PType const & p = aos_ptr[i];
-                        amrex::ParticleReal const x = p.pos(0);
-                        amrex::ParticleReal const y = p.pos(1);
-                        amrex::ParticleReal const t = p.pos(2);
+                                // access AoS data such as positions and cpu/id
+                                PType const &p = aos_ptr[i];
+                                amrex::ParticleReal const x = p.pos(0);
+                                amrex::ParticleReal const y = p.pos(1);
+                                amrex::ParticleReal const t = p.pos(2);
 
-                        // access SoA Real data
-                        amrex::ParticleReal const px = part_px[i];
-                        amrex::ParticleReal const py = part_py[i];
-                        amrex::ParticleReal const pt = part_pt[i];
+                                // access SoA Real data
+                                amrex::ParticleReal const px = part_px[i];
+                                amrex::ParticleReal const py = part_py[i];
+                                amrex::ParticleReal const pt = part_pt[i];
 
-                        // write particle data to file
-                        amrex::PrintToFile(file_name)
-                            << x << " " << y << " "<< t << " "
-                            << px << " " << py << " " << pt << "\n";
-                    } // i=0...np
-                } // if( otype == OutputType::PrintParticles)
-            } // end loop over all particle boxes
-        } // env mesh-refinement level loop
+                                // write particle data to file
+                                amrex::PrintToFile(file_name)
```

Comment:
```suggestion
                                amrex::AllPrintToFile(file_name)
```

`PrintToFile` will still only do anything on the IO rank

Reply:
oh, ok. let me check.

Reply:
We could also generate N files for now via `AllPrintToFile` and adjust the post-processing scripts for now to use a glob.

Reply:
Done that way now, pandas is powerful enough to handle this :)